packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -139,6 +146,16 @@ export default util.createRule<Options, MessageIds>({
       }
       return false;
     }
+
+    function isIIFE(
+      node: TSESTree.ArrowFunctionExpression | TSESTree.FunctionExpression,
+    ): boolean {
+      return (
+        node.parent?.type === AST_NODE_TYPES.CallExpression &&
+        node.parent?.callee === node
```

Comment:
If `node.parent.type === AST_NODE_TYPES.CallExpression` then we know `node.parent` exists:

```suggestion
        node.parent.callee === node
```